internal/process/steps/gardener_cluster.go

```diff
@@ -0,0 +1,134 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewSyncGardenerCluster(os storage.Operations, k8sClient client.Client) syncGardenerCluster {
+	return syncGardenerCluster{
+		k8sClient:        k8sClient,
+		operationManager: process.NewOperationManager(os),
+	}
+}
+
+type syncGardenerCluster struct {
+	k8sClient        client.Client
+	operationManager *process.OperationManager
+}
+
+func (_ *syncGardenerCluster) Name() string {
+	return "Sync_GardenerCluster"
+}
+
+func (s *syncGardenerCluster) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	gardenerCluster, err := s.GetOrCreateNewGardenerCluster(operation.RuntimeID, operation.KymaResourceNamespace)
+	if err != nil {
+		log.Errorf("unable to get GardenerCluster %s/%s", operation.KymaResourceNamespace, operation.RuntimeID)
+		return s.operationManager.RetryOperation(operation, "unable to get GardenerCluster", err, 3*time.Second, 20*time.Second, log)
+	}
+	gardenerCluster.SetShootName(operation.ShootName)
+	gardenerCluster.SetKubecofigSecret(fmt.Sprintf("kubeconfig-%s", operation.RuntimeID), operation.KymaResourceNamespace)
+
+	obj := gardenerCluster.ToUnstructured()
+	ApplyLabelsAndAnnotationsForLM(obj, operation)
+
+	if gardenerCluster.ExistsInTheCluster() {
+		err = s.k8sClient.Update(context.Background(), obj)
+	} else {
+		err = s.k8sClient.Create(context.Background(), obj)
+	}
+	if err != nil {
+		log.Errorf("unable to apply GardenerCluster %s/%s", operation.KymaResourceNamespace, operation.RuntimeID)
```

Comment:
If we log the message here it would be harder to determine whether we executed `Update` or `Create`